src/rules/no-unused-modules.js

```diff
@@ -0,0 +1,731 @@
+/**
+ * @fileOverview Ensures that modules contain exports and/or all 
+ * modules are consumed within other modules.
+ * @author René Fermann
+ */
+
+import Exports from '../ExportMap'
+import resolve from 'eslint-module-utils/resolve'
+import docsUrl from '../docsUrl'
+
+// eslint/lib/util/glob-util has been moved to eslint/lib/util/glob-utils with version 5.3
+let listFilesToProcess
+try {
+  listFilesToProcess = require('eslint/lib/util/glob-utils').listFilesToProcess
+} catch (err) {
+  listFilesToProcess = require('eslint/lib/util/glob-util').listFilesToProcess
+}
+
+const EXPORT_DEFAULT_DECLARATION = 'ExportDefaultDeclaration'
+const EXPORT_NAMED_DECLARATION = 'ExportNamedDeclaration'
+const EXPORT_ALL_DECLARATION = 'ExportAllDeclaration'
+const IMPORT_DECLARATION = 'ImportDeclaration' 
+const IMPORT_NAMESPACE_SPECIFIER = 'ImportNamespaceSpecifier'
+const IMPORT_DEFAULT_SPECIFIER = 'ImportDefaultSpecifier' 
+const VARIABLE_DECLARATION = 'VariableDeclaration'
+const FUNCTION_DECLARATION = 'FunctionDeclaration'
+const DEFAULT = 'default'
+
+let preparationDone = false
+const importList = new Map()
+const exportList = new Map()
+const ignoredFiles = new Set()
+
+const isNodeModule = path => {
+  return path.includes('node_modules')
```

Comment:
should this ensure it's `(^|\/)node_modules\/|`?

Reply:
I'm not sure, if I understand that question correctly. What's the background of this question?

Reply:
A folder named `backup_node_modules` shouldn't return true here, nor should `node_modules_old`. Only a path that is *inside* a directory called, verbatim, `node_modules`, should get a `true`.

Reply:
That would mean, if someone makes a backup of the ```node_modules``` folder and renames the new folder to e.g. ```node_modules_old```, all files within this new folder will be analyzed and perhaps reported, if ```src``` and ```ignoreExports```aren't explicitly defined.

But on the other side it's not up to us covering strange folder naming, is it? I will adjust it later.

Reply:
Yes, that’s exactly right - if they do that, they should see reports until they explicitly ignore the new folder.